### PR TITLE
style: unify Modal and Drawer close icon style

### LINKED ISSUE #47890: Modal 关闭按钮可点击区域比较小
### Reproduction link

[https://ant.design/components/modal-cn](https://ant.design/components/modal-cn)

### Steps to reproduce

打开一个带 X 按钮的 Modal

### What is expected?

关闭按钮的实际可点击区域应该和 Button （仅图标）类似比较大，32x32

image

### What is actually happening?

关闭按钮的实际可点击区域只有 22x22px

image

| Environment | Info |
| --- | --- |
| antd | 5.15.2 |
| React | 18.2.0 |
| System | macOS |
| Browser | Chrome 121 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/47890

### 💡 Background and solution

图片

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Unify Modal, Drawer, Tour and notification close button style.         |
| 🇨🇳 Chinese |  统一 Modal、Drawer、Tour、notification 关闭按钮的样式，增大关闭按钮的操作区域使其更容易点击。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/drawer/style/index.ts [hunk 1]
```diff
@@ -20,13 +20,18 @@ export interface ComponentToken {
    * @descEN Horizontal padding of footer
    */
   footerPaddingInline: number;
+  /** @internal */
```
Comment (crazyair) [inline components/drawer/style/index.ts:23]: 为什么是内部的？

Comment (afc163) [reply components/drawer/style/index.ts:23]: 这个 internal 很奇怪，应该直接放下面的 DrawerToken 里，我先来个 PR 清理一下。

**Review by li-jia-nan — DISMISSED** [review]

**Review by crazyair — DISMISSED** [review]

File: components/drawer/style/index.ts [hunk 2]
```diff
@@ -181,9 +190,17 @@ const genDrawerStyle: GenerateStyle<DrawerToken> = (token) => {
...
+        transition: `all ${motionDurationMid}`,
         textRendering: 'auto',
 
+        '&:hover': {
+          backgroundColor: colorBgTextHover,
+        },
+
+        '&:active': {
+          backgroundColor: colorBgTextActive,
+        },
```
Comment (afc163) [inline components/drawer/style/index.ts:202]: 直接用和 Button text 一致的色彩就够了。

File: components/modal/style/index.ts [hunk 3]
```diff
@@ -367,8 +363,6 @@ export const prepareComponentToken = (token: GlobalToken) => ({
   titleColor: token.colorTextHeading,
 
   // internal
-  closeBtnHoverBg: token.wireframe ? 'transparent' : token.colorFillContent,
-  closeBtnActiveBg: token.wireframe ? 'transparent' : token.colorFillContentHover,
```
Comment (afc163) [inline components/modal/style/index.ts:371]: hover 的背景色不会影响是否是 wireframe 线框风格，因此简单点就好了，不重复提取 closeBtnHoverBg 和 closeBtnActiveBg 内部变量。

**Review by li-jia-nan — DISMISSED** [review]

File: components/notification/PurePanel.tsx [hunk 4]
```diff
@@ -27,13 +27,7 @@ export function getCloseIcon(prefixCls: string, closeIcon?: React.ReactNode): Re
...
     return null;
   }
-  return (
-    closeIcon || (
-      <span className={`${prefixCls}-close-x`}>
-        <CloseOutlined className={`${prefixCls}-close-icon`} />
-      </span>
-    )
-  );
+  return closeIcon || <CloseOutlined className={`${prefixCls}-close-icon`} />;
```
Comment (yoyo837) [inline components/notification/PurePanel.tsx:30]: 确定不走 feature吗

Comment (afc163) [reply components/notification/PurePanel.tsx:30]: 这个对用户几乎无影响

Comment (crazyair) [reply components/notification/PurePanel.tsx:30]: 有些加功能确实无感，有些修 bug 还会带出其他 bug，没法界定

File: components/drawer/style/index.ts [hunk 5]
```diff
@@ -173,12 +174,12 @@ const genDrawerStyle: GenerateStyle<DrawerToken> = (token) => {
 
       [`${componentCls}-close`]: {
         display: 'inline-flex',
-        width: controlHeight,
-        height: controlHeight,
+        width: calc(fontSizeLG).add(paddingXS).equal(),
+        height: calc(fontSizeLG).add(paddingXS).equal(),
```
Comment (afc163) [inline components/drawer/style/index.ts:178]: Drawer 不能和 Modal 的关闭按钮一样大，会影响 Drawer Header 的高度。这里改小回来。

**Review by li-jia-nan — APPROVED** [review]

**Comment by afc163** [comment]
Virsual Diff 的内容混掉了，真正的 diff 在这里 https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-47909/visualRegressionReport/report.html

**Comment by crazyair** [comment]
我是把diff评论删了，重新提个空提交

**Pull request merged**